tenseal/tensors/ckksvector.cpp

```diff
@@ -200,4 +201,44 @@ CKKSVector& CKKSVector::mul_plain_inplace(vector<double> to_mul) {
     return *this;
 }
 
+CKKSVector CKKSVector::matmul_right(const vector<vector<double>> matrix) {
+    // matrix is organized by rows
+    // _check_matrix(matrix, this->size())
+    size_t n_rows = matrix.size();
+    size_t n_cols = matrix[0].size();
+
+    if (this->size() != matrix.size()) {
+        cout << this->size() << " vs " << n_rows << endl;
+        throw invalid_argument("matrix shape doesn't match with vector size");
+    }
+
+    CKKSVector result_vector = *this;
+    auto encoder = this->context->get_encoder<CKKSEncoder>();
+
+    Ciphertext ct;
```

Comment:
Maybe is better to declare the ct, pt_diag and diag in the loop scope, to prevent accidental use after that.